sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyPipelinePolicy.cs

```diff
@@ -99,18 +101,24 @@ private static void BuildCanonicalizedHeaders(StringBuilder stringBuilder, HttpM
         {
             // Grab all the "x-ms-*" headers, trim whitespace, lowercase, sort,
             // and combine them with their values (separated by a colon).
-            foreach (var header in
-                message.Request.Headers
-                .Where(static h => h.Name.StartsWith(Constants.HeaderNames.XMsPrefix, StringComparison.OrdinalIgnoreCase))
-#pragma warning disable CA1308 // Normalize strings to uppercase
-                .Select(static h => (h.Name.ToLowerInvariant(), h.Value))
-#pragma warning restore CA1308 // Normalize strings to uppercase
-                .OrderBy(static h => h.Item1.Trim()))
+            var headers = new List<HttpHeader>();
+            foreach (var header in message.Request.Headers)
+            {
+                if (header.Name.StartsWith(Constants.HeaderNames.XMsPrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    headers.Add(header);
+                }
+            }
+
+            headers.Sort(static (x, y) => string.CompareOrdinal(x.Name, y.Name));
```

Comment:
Is this refactor of the sorting and enumeration needed below in `BuildCanonicalizedResource` also?

Reply:
Maybe. `BuildCanonicalizedResource` is not visible in the profiler, probably because I don't have that many query parameters in test.